crypto/crypto.lurk

```diff
@@ -0,0 +1,96 @@
+; crypto.lurk
+; 
+; simple crypto-related functions
+;
+; load in repl as:
+; > :load "crypto.lurk"
+
+
+(letrec (
+
+    ; parity tests
+    (odd (lambda (a) (< (/ a 2) 0)))
+    (even (lambda (a) (eq NIL (odd a))))
+
+    ; square & multiply exponentiation
+    (fastexp (lambda (b e)
+        (if (= e 0) 1
+            (if (even e) (fastexp (* b b) (/ e 2))
+                (* b (fastexp (* b b) (/ (- e 1) 2)))))))
+
+    ; GCD
+    (gcd (lambda (a b) 
+        (if (= a b) a
+            (if (> a b) (gcd (- a b) b)
+                (gcd (- b a) a)))))
```

Comment:
Neat. Note that GCD only makes sense if you view `a` and `b` as natural numbers. (As field elements, you could kind of argue that _every_ value is divided by <whichever value you consider 'greatest'>). If that's how you mean it, then it's important to specify (or check) that `a` and `b` are in range. I think if either one is negative, then your `>` check won't give the right results. We spent some time thinking about this example  (GCD — not your code) recently, but I still might have gotten a detail wrong in my quick summary. cc: @emmorais @differentialderek @weissjeffm

Reply:
Consider adding a comment to clarify, so people studying this code don't get confused or misled.

Reply:
This is sleek—the nice thing about it is that it doesn't require values greater than `a` and `b` so from the "field-elements-as-integers" this is simulates normal gcd for positives.

I have a *much more clunky* version here: https://github.com/lurk-lang/lurk-lib/pull/32